helix-term/src/ui/completion.rs

```diff
@@ -33,11 +33,57 @@ impl menu::Item for CompletionItem {
             .into()
     }
 
-    fn format(&self, _data: &Self::Data) -> menu::Row {
+    // fn label(&self, _data: &Self::Data) -> Spans {
+    //     self.label.as_str().into()
+    // }
+
+    fn format(&self, _data: &Self::Data, theme: Option<&Theme>) -> menu::Row {
+        let (lsp_type_label, style) = match self.kind {
+            Some(lsp::CompletionItemKind::TEXT) => ("text", Some("ui.text")),
+            Some(lsp::CompletionItemKind::METHOD) => ("method", Some("function.method")),
+            Some(lsp::CompletionItemKind::FUNCTION) => ("function", Some("function")),
+            Some(lsp::CompletionItemKind::CONSTRUCTOR) => ("constructor", Some("constructor")),
+            Some(lsp::CompletionItemKind::FIELD) => ("field", Some("variable.other.member")),
+            Some(lsp::CompletionItemKind::VARIABLE) => ("variable", Some("variable")),
+            Some(lsp::CompletionItemKind::CLASS) => ("class", Some("type")),
+            Some(lsp::CompletionItemKind::INTERFACE) => ("interface", Some("type")),
+            Some(lsp::CompletionItemKind::MODULE) => ("module", Some("module")),
+            Some(lsp::CompletionItemKind::PROPERTY) => ("property", Some("attributes")),
+            Some(lsp::CompletionItemKind::UNIT) => ("unit", Some("constant")),
+            Some(lsp::CompletionItemKind::VALUE) => ("value", Some("string")),
+            Some(lsp::CompletionItemKind::ENUM) => ("enum", Some("type")),
+            Some(lsp::CompletionItemKind::KEYWORD) => ("keyword", Some("keyword")),
+            Some(lsp::CompletionItemKind::SNIPPET) => ("snippet", None),
+            Some(lsp::CompletionItemKind::COLOR) => ("color", None),
+            Some(lsp::CompletionItemKind::FILE) => ("file", None),
+            Some(lsp::CompletionItemKind::REFERENCE) => ("reference", None),
+            Some(lsp::CompletionItemKind::FOLDER) => ("folder", None),
+            Some(lsp::CompletionItemKind::ENUM_MEMBER) => {
+                ("enum_member", Some("type.enum.variant"))
+            }
+            Some(lsp::CompletionItemKind::CONSTANT) => ("constant", Some("constant")),
+            Some(lsp::CompletionItemKind::STRUCT) => ("struct", Some("type")),
+            Some(lsp::CompletionItemKind::EVENT) => ("event", None),
+            Some(lsp::CompletionItemKind::OPERATOR) => ("operator", Some("operator")),
+            Some(lsp::CompletionItemKind::TYPE_PARAMETER) => {
+                ("type_param", Some("function.parameter"))
+            }
+            Some(kind) => unimplemented!("{:?}", kind),
+            None => ("", None),
+        };
+        let mut lsp_type_style = theme
+            .zip(style)
+            .map(|(theme, style)| theme.get(style))
+            .unwrap_or_default()
+            .remove_modifier(Modifier::all())
+            .add_modifier(Modifier::ITALIC);
```

Comment:
Why are we intentionally removing the style from the theme and hardcoding italics? What's the point of pulling in user themes if we're just going to ignore them? I personally can't stand italics in code.

Reply:
We are pulling the theme to apply the colors. This is for the completions items, they should have a matching modifier, i.e. no bold but all in italic. Please have a look at the screenshots above.

Reply:
Yeah, in my opinion, we should just let the theme decide how to style the completion items here and not hard code style for the user.